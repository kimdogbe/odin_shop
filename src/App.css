* {
  box-sizing: border-box;
}

#root {
  margin: 0 auto;
  text-align: center;
  width: 100vw;
}

.logo {
  height: 6em;
  padding: 1.5em;
  will-change: filter;
  transition: filter 300ms;
}
.logo:hover {
  filter: drop-shadow(0 0 2em #646cffaa);
}
.logo.react:hover {
  filter: drop-shadow(0 0 2em #61dafbaa);
}

@keyframes logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

@media (prefers-reduced-motion: no-preference) {
  a:nth-of-type(2) .logo {
    animation: logo-spin infinite 20s linear;
  }
}

.card {
  padding: 2em;
}

.read-the-docs {
  color: #888;
}

nav ul{
  display: flex;
  justify-content: space-between;
  align-items: center;
  background-color: #8888881f;
  width: 100%;
  padding: 0em 1em;
}

nav ul li {
  list-style: none;
  padding: 10px;
}

#home-btn {
  margin-right: auto;
}