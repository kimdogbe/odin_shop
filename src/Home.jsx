import ShopButton from "./components/ShopBtn"

function Home() {
  return (
    <div id="home-section">
      <h1>Hello and welcome to our store!</h1>
      <p>We hope you find what you're looking for</p>
      <ShopButton />
    </div>
  )
}

export default Home