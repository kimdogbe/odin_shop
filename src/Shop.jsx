import { Link } from "react-router-dom"
import ProductCard from "./components/ProductCard"

function Shop() {
  return (
    <div>
      <h1>Shop page</h1>
      <p>Browse from our products below!</p>
      <ProductCard />
    </div>
  )
}

export default Shop