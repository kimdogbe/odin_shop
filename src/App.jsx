import './App.css'
import { Link, Outlet } from 'react-router-dom'

function App() {
  return (
    <div>
      <nav>
        <ul>
          <h2>Kays Shop</h2>
          <li id='home-btn'>
            <Link to="home">Home page</Link>
          </li>
          <li>
            <Link to="about">About</Link>
          </li>
          <li>
            <Link to="checkout">Checkout</Link>
          </li>
        </ul>
      </nav>
      <Outlet />
    </div>
  )
}

export default App
