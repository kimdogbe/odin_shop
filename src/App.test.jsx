import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import { BrowserRouter, RouterProvider } from 'react-router-dom';
import App from './App';
import Home from './Home';
import Shop from './Shop';
import userEvent from '@testing-library/user-event';

describe('App component', () => {
  it('renders main logo', () => {
    render(<App />, {wrapper: BrowserRouter});
    expect(screen.getByRole('heading').textContent).toMatch("Kays Shop");
  });

  it('renders all links', () => {
    render(<App />, {wrapper: BrowserRouter});
    expect(screen.getAllByRole('link')).toHaveLength(4);
  });

  it('renders correct link names and paths', () => {
    render(<App />, {wrapper: BrowserRouter});
    expect(screen.getAllByRole('link').map(link => link.textContent)).toContain('About');
    expect(screen.getAllByRole('link').map(link => link.textContent)).toContain('Shop');
    expect(screen.getAllByRole('link').map(link => link.textContent)).toContain('Checkout');
  });

  it('app snapshot', () => {
    const { container } = render(<App />, {wrapper: BrowserRouter});
    expect(container).toMatchSnapshot();

  });

});

describe('Home component', () => {
  it('renders heading and text', () => {
    render(<Home />, {wrapper: BrowserRouter});
    
    expect(screen.getByRole("heading").textContent).toMatch(/Hello and welcome to our store/i);
    expect(screen.getByRole("paragraph").textContent).toMatch(/We hope you find what you're looking for/i);
  });

  it('home page snapshot', () => {
    const { container } = render(<Home />, {wrapper: BrowserRouter});
    expect(container).toMatchSnapshot();
  });
});


// describe('Shop component', () => {  
//   it('renders heading and text', () => {
//     render(<Shop context={[]} />, {wrapper: BrowserRouter});
    
//     // expect(screen.getByRole("heading").textContent).toMatch(/shop page/i);
//     expect(screen.getByRole("paragraph").textContent).toMatch(/hello and welcome to our store!/i);
//   })
// })