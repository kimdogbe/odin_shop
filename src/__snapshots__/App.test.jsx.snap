// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > app snapshot 1`] = `
<div>
  <div>
    <nav>
      <ul>
        <a
          data-discover="true"
          href="/"
          id="home-logo"
        >
          <h2>
            Kays Shop
          </h2>
        </a>
        <li>
          <a
            data-discover="true"
            href="/about"
          >
            About
          </a>
        </li>
        <li>
          <a
            data-discover="true"
            href="/shop"
          >
            Shop
          </a>
        </li>
        <li>
          <a
            data-discover="true"
            href="/checkout"
          >
            Checkout
          </a>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;

exports[`Home component > home page snapshot 1`] = `
<div>
  <div>
    <h1>
      Hello and welcome to our store!
    </h1>
    <p>
      We hope you find what you're looking for
    </p>
    <a
      data-discover="true"
      href="/shop"
    >
       
      <button>
        Shop
      </button>
    </a>
  </div>
</div>
`;
